rbx_binary/src/deserializer/state.rs

```diff
@@ -440,11 +441,23 @@ This may cause unexpected or broken behavior in your final results if you rely o
                         add_property(instance, &property, value.as_ref().into());
                     }
                 }
+                VariantType::ContentId => {
+                    for referent in &type_info.referents {
+                        let instance = self.instances_by_ref.get_mut(referent).unwrap();
+                        let value = chunk.read_string()?;
+                        add_property(instance, &property, ContentId::from(value).into());
+                    }
+                }
                 VariantType::Content => {
                     for referent in &type_info.referents {
                         let instance = self.instances_by_ref.get_mut(referent).unwrap();
-                        let value: Content = chunk.read_string()?.into();
-                        add_property(instance, &property, value.into());
+                        let value = chunk.read_string()?;
+                        let content = if value.is_empty() {
+                            Content::none()
+                        } else {
+                            Content::from_uri(value)
+                        };
+                        add_property(instance, &property, content.into());
                     }
```

Comment:
Does this arm need to be here? Are there any properties that have a serialized type of `String` with a canonical type of `Content`?

Reply:
Legacy `ContentId` uses `string` serialization for binary format, but we don't support `string` for new `Content` properties.

Reply:
None that I know of! I can just remove this; it was part of me being on autopilot and replacing various parts of `ContentId` with `Content`.